analyzers/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -160,6 +189,31 @@ private static IEnumerable<XElement> ParseXmlSettings(SourceText sourceText)
             }
         }
 
+        private static bool IsTestProject(TryGetValueDelegate<ProjectConfigReader> tryGetValue, Compilation c, AnalyzerOptions options)
+        {
+            ProjectType projectType = ProjectType.None;
+            if (options.AdditionalFiles.FirstOrDefault(IsSonarProjectConfig) is { } sonarProjectConfigXml)
+            {
+                var projectConfigReader = sonarProjectConfigXml.GetText() is { } sourceText
+                    // TryGetValue catches all exceptions from SourceTextValueProvider and returns false when thrown
+                    && tryGetValue(sourceText, ProjectConfigProvider, out var cachedProjectConfigReader)
+                    ? cachedProjectConfigReader
+                    : throw new InvalidOperationException($"File {Path.GetFileName(sonarProjectConfigXml.Path)} has been added as an AdditionalFile but could not be read and parsed.");
```

Comment:
This whole section should be replaced with something like
```
var config = ProjectConfiguration(options);
if(config.IsScanner){

}
```
I was tempted to add something like `IsScanner => !string.IsNullOrEmpty(OutPath)` but there weren't enough usages before (I refactored the CBDE one differently). 

I'm not very happy with `IsScanner` name, but it might inspire you :)

Reply:
That should simplify the whole logic to
```
return config.IsScanner 
    ? config.... 
    :  c?.IsTest() ?? true
```

Reply:
I will do this after I get the 2nd round of reviews for the rest of the PR